garden-backend-service/src/models/mdf_dataset.py

```diff
@@ -0,0 +1,20 @@
+from sqlalchemy import ForeignKey, String
+from sqlalchemy.dialects.postgresql import ARRAY
+from sqlalchemy.orm import Mapped, mapped_column
+from src.models.base import Base
+
+
+class Dataset(Base):
+    __tablename__ = "mdf-datasets"
```

Comment:
pretty arbitrary but we've been snake_casing our table names so far
```suggestion
    __tablename__ = "mdf_datasets"
```